ironic-python-agent/0002-Run-system_installer-functions.patch

```diff
@@ -0,0 +1,51 @@
+From cac108c6d35ffe9283265c2e65a3ea733683d792 Mon Sep 17 00:00:00 2001
+From: "Grzegorz Grasza (xek)" <grzegorz.grasza@intel.com>
+Date: Thu, 4 Jan 2018 13:55:10 +0100
+Subject: [PATCH 2/2] Run system_installer functions
+
+---
+ ironic_python_agent/extensions/standby.py | 21 +++++++++++++++++++++
+ 1 file changed, 21 insertions(+)
+
+diff --git a/ironic_python_agent/extensions/standby.py b/ironic_python_agent/extensions/standby.py
+index 1b9f2db..bd4e79d 100644
+--- a/ironic_python_agent/extensions/standby.py
++++ b/ironic_python_agent/extensions/standby.py
+@@ -23,6 +23,7 @@ from oslo_config import cfg
+ from oslo_log import log
+ 
+ from ironic_lib import disk_utils
++from ironic_lib import system_installer
+ from ironic_python_agent import errors
+ from ironic_python_agent.extensions import base
+ from ironic_python_agent import hardware
+@@ -451,6 +452,26 @@ class StandbyExtension(base.BaseAgentExtension):
+              large to store on the given device.
+         """
+         LOG.debug('Preparing image %s', image_info['id'])
++
++        disk_config = self.agent.node['instance_info'].get('disk_config')
++        if disk_config:
++            LOG.debug('Found disk_config in instance_info,'
++                      ' running DiskPartitioner install...')
++            assert image_info.get('disk_format') == 'qcow2'
```

Comment:
Can we raise a proper exception here? :)

Reply:
I left that here just in case, it will raise an exception while trying to mount the image vie guestfs. Maybe just remove it?

Reply:
Sure, I'm fine with checking it, but would rather do something like:

```
disk_format = image_info.get('disk_format')
if disk_format  != 'qcow2':
    raise ImageFormatError('Image format is %s, must be qcow2 when using disk_config' % disk_format)
```

Exceptions raised here will automagically be propagated back to node.last_error and such, we should try to make the error as clear as possible for the operator.